cmake/find/FindJulia.cmake

```diff
@@ -0,0 +1,108 @@
+macro(julia_bail_if_false message var)
+  if(NOT ${var})
+    set(Julia_FOUND 0)
+    set(Julia_NOT_FOUND_MESSAGE "${message}")
+    return()
+  endif()
+endmacro()
+
+# https://docs.julialang.org/en/v1/manual/environment-variables/#JULIA_BINDIR
+find_program(JULIA_EXECUTABLE julia PATHS ENV JULIA_BINDIR)
+julia_bail_if_false("The Julia executable could not be found" JULIA_EXECUTABLE)
+
+if(NOT DEFINED JULIA_BINDIR)
+  # The executable could be a chocolatey shim, so run some Julia code to report
+  # the path of the BINDIR
+  execute_process(
+      COMMAND "${JULIA_EXECUTABLE}" -e "print(Sys.BINDIR)"
+      OUTPUT_VARIABLE julia_bindir
+  )
+  file(TO_CMAKE_PATH "${julia_bindir}" julia_bindir)
+  set(JULIA_BINDIR "${julia_bindir}" CACHE PATH "")
+endif()
+julia_bail_if_false("The Julia executable could not report its location" JULIA_BINDIR)
```

Comment:
```suggestion
if (NOT DEFINED JULIA_BINDIR OR "${JULIA_BINDIR}" STREQUAL "")

    # try through command line instead
    execute_process(
        COMMAND "julia -e \"print(Sys.BINDIR)\""
        OUTPUT_VARIABLE JULIA_BINDIR_LOCAL
    )

    if ("${JULIA_BINDIR_LOCAL}" STREQUAL "")
        # if command line fails, fall back to user specified JULIA_PATH
        if (NOT DEFINED ENV{JULIA_PATH})
         message(FATAL_ERROR "Unable to automatically determine location of julia image. Before running cmake, please manually set the environment variable JULIA_BINDIR using\n    export JULIA_PATH=/path/to/your/.../julia/bin\nIf you are unsure of the location of your julia images binary path, you can access the path from within the julia REPL using\n    println(Sys.BINDIR)
    For more information, visit https://github.com/Clemapfel/jluna/blob/master/README.md#troubleshooting")
        else()
            set(JULIA_BINDIR_LOCAL $ENV{JULIA_PATH})
        endif()
    endif()

    file(TO_CMAKE_PATH ${JULIA_BINDIR_LOCAL} JULIA_BINDIR_LOCAL)
    set(JULIA_BINDIR ${JULIA_BINDIR_LOCAL})
endif()
```
Let me know if the `STREQUAL ""` is usually necessary, it was on my system, cmake treated JULIA_BINDIR as defined but it has no value

Reply:
Also I added a fall-back `JULIA_PATH`, if the automatic detection fails its much easier to explain to users to just set the environment variable, rather than ask them to fix the julia install. For most users this works automatically without any interaction anyway

Reply:
Where did you get `JULIA_PATH` from? I only found `JULINA_BINDIR` from the documentation. These are also cache variables, the user can just provide them manually on the configure line: `cmake -B build "-DJULIA_BINDIR=c:/users/REDACTED/appdata/local/programs/Julia-1.7.2/bin" ...`

Just an FYI, but the CMake code suggestion has many errors:
- `if (NOT DEFINED JULIA_BINDIR OR "${JULIA_BINDIR}" STREQUAL "")`: `"${JULIA_BINDIR}"` is unnecessarily expanded, should be just `JULIA_BINDIR`
- `COMMAND "julia -e \"print(Sys.BINDIR)\""`: this says that `julia -e "print(Sys.BINDIR)"` is the executable's name
- `if ("${JULIA_BINDIR_LOCAL}" STREQUAL "")`: potential undefined variable use, just `JULIA_BINDIR_LOCAL` is enough
- `$ENV{JULIA_PATH}` and `${JULIA_BINDIR_LOCAL}`: unquoted variable use is a list expansion, just like in Bash